system/lib/pthread/emscripten_thread_init.c

```diff
@@ -7,13 +7,18 @@
 #include <pthread.h>
 #include "emscripten/threading.h"
 #include "threading_internal.h"
+#include "pthread_impl.h"
 
 void _emscripten_thread_init(pthread_t ptr,
                              int is_main,
                              int is_runtime,
                              int can_block,
+                             int default_stacksize,
                              int start_profiling) {
   __set_thread_state(ptr, is_main, is_runtime, can_block);
+  if (is_main && default_stacksize) {
+    __default_stacksize = default_stacksize;
```

Comment:
I think just doing `__default_stacksize = default_stacksize` save on complexity and code size... I would also worry that this might clobber other default attribtues other than stack size.

Also, I think the clamping of DEFAULT_PTHREAD_STACK_SIZE can be done in the compiler front end where options are generally checked.  No need to do it at runtime too.

Reply:
I see. Perhaps limiting `-sDEFAULT_PTHREAD_STACK_SIZE` isn't necessary at all? That 8Mb limit is probably Unix specific. Though for Wasm, it might make sense to introduce that as an upper limit.